### PR TITLE
MH-12607 (Harvard DCE), Lossless Concat Operation

### PR DESCRIPTION
**Pull request by lkiesow**
This is an updated version of pull request #14 which @phoebem updated [here](https://github.com/lkiesow/opencast/pull/1). During my review of #14, I noticed a few issues for which I provided some patches which are included here.

To make the review easier, this pull request contains only the lossless concat parts of #14.

### CODE REVIEW

**Comment by lkiesow** [comment]
- @gregorydlogan, I already included fixes based on your review from yesterday.
- @phoebem, can you check if this works for you too.

**Comment by staubesv** [comment]
As discussed with @lkiesow and @gregorydlogan, I'm taking over this review.

File: docs/guides/admin/docs/workflowoperationhandlers/concat-woh.md [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
---------
+The *concat* operation handler has been created to concatenate multiple video tracks into one video track.
 
-The "concat" operation handler has been created to concatenate multiple video tracks into one video track. Note, that
-the concatenation process used will always re-encode the videos.
+For a concatenation of two video files to work, both files need to have the same format (timebase, resolution, codecs,
+frame rate, etc.). This workflow operation has two modes to deal with this restriction:
+
+- A *general* mode which re-encodes all input files, hence ensuring that this restriction is always met.
+- A *same codec* mode which assumes the restriction is already met and can hence concatinate the files much faster while
```
Comment (staubesv) [inline docs/guides/admin/docs/workflowoperationhandlers/concat-woh.md]: concat**e**nate

File: modules/composer-ffmpeg/src/main/java/org/opencastproject/composer/impl/ComposerServiceImpl.java [hunk 2]
```diff
@@ -721,8 +724,13 @@ public Job concat(String profileId, Dimension outputDimension, float outputFrame
       arguments.add(1, "");
     }
     arguments.add(2, String.format(Locale.US, "%f", outputFrameRate));
+    if (sameCodec) {
+      arguments.add(3, "true");
+    } else {
+      arguments.add(3, "");
```
Comment (staubesv) [inline modules/composer-ffmpeg/src/main/java/org/opencastproject/composer/impl/ComposerServiceImpl.java]: Why not "false"?

**Review by staubesv — CHANGES_REQUESTED** [review]
Hi @lkiesow 

Just finished a first round of review. Haven't found any major issues, just some minor comments.

**Comment by lkiesow** [comment]
@staubesv, This should fix both issues you raised.
Please wait with the merge though. I like to rebase this to ensure the new Markdown linter does not complain. I'll make that update later today. That should not keep you from testing though ;-)

**Comment by lkiesow** [comment]
Rebased this to the head of develop to include all of the updated style and formatting requirements and tests. Let's wait for the build to finish though I hopefully got all of them.

**Comment by lkiesow** [comment]
Finally seems like I've got all errors fixed.
There had also been some Javadoc errors.
Should be ready to be merged now.

**Review by staubesv — APPROVED** [review]

**Pull request merged**